public/react/SettingsSidebar/SettingsSidebar.tsx

```diff
@@ -31,45 +36,43 @@ const SettingsSidebar: FC<SettingsSidebarProps> = ({
   const derivedIssuesObservable: CanObservable<{ status: string; team: { name: string } }[]> =
     value.from(routeData, "derivedIssues");
 
+  const [settingVisible, setSettingVisible] = useState(false);
+  const settingsWidth = showSettings ? SETTINGS_WIDTH[showSettings] : "w-0";
+
   const changeSettings = (settings = "") => {
     routeData.showSettings = settings;
   };
-
   const returnToSettings = () => changeSettings("");
-
   return (
     <div className="px-6 py-2 h-full min-w-40">
       {showSidebarBranding && <Branding />}
       {!showSettings && <ReportSettings changeSettings={changeSettings} />}
       {!!showSettings && showSettings !== "TEAMS" && (
         <GoBackButton hideSettings={returnToSettings} />
       )}
-      {showSettings === "SOURCES" && (
-        <div className="w-96">
-          <IssueSource />
-        </div>
-      )}
-      {showSettings === "TIMING" && <TimingCalculation />}
-      {showSettings === "TEAMS" && (
-        <div className="h-full">
-          <TeamConfiguration
-            derivedIssuesObservable={derivedIssuesObservable}
-            showSidebarBranding={showSidebarBranding}
-            onUpdate={onUpdateTeamsConfiguration}
-            onBackButtonClicked={() => returnToSettings()}
-          />
-        </div>
-      )}
-      {showSettings === "THEME" && (
-        <div className="w-80 h-full">
-          <Theme onBackButtonClicked={changeSettings} />
-        </div>
-      )}
-      {showSettings === "REPORTS" && (
-        <div className="h-full">
-          <ViewReports onBackButtonClicked={changeSettings} />
-        </div>
-      )}
+      <div
+        className={`overflow-hidden h-full transition-[width] duration-500 ease-in-out ${settingsWidth}`}
+        onTransitionEnd={() => {
+          setSettingVisible(!!showSettings);
```

Comment:
In order to handle the setting content not looking weird as the transition is happening. A new flag is used to determine when the transition ends before displaying the content